### PR TITLE
[Recure]: added cookies saver for visitorId/refactor some unnecessary…

### PR DESCRIPTION
**Pull request by Abnormaltype**
added cookies saver for visitorId/refactor some unnecessary imports

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -14,22 +15,20 @@ export enum EventType {
...
 }
 
-export type Payload = {
+type Payload = {
   userId: string;
   provider: string;
   type: EventType;
   visitorId: string;
-  visitorFound: boolean;
-  confidence: object;
```
Comment (danylott) [inline src/index.ts:23]: I'd leave them, if not a problem

File: src/index.ts [hunk 2]
```diff
@@ -47,36 +46,84 @@ export async function getApiKey(
   return data.apiKey;
 }
 
+function setCookie(cookieName: string, cookieValue: string, exdays: number) {
+  const date = new Date();
+  date.setTime(date.getTime() + (exdays*24*60*60*1000));
+  const expires: string = "expires="+ date.toUTCString();
+  document.cookie = `${cookieName}=${cookieValue};${expires};path=/`;
```
Comment (danylott) [inline src/index.ts:53]: So we need direct access to the document here?
@ArtemTopolskyi could you help here, is it a bad practice?

File: src/index.ts [hunk 3]
```diff
@@ -47,36 +46,84 @@ export async function getApiKey(
...
+}
+
+function getCookie(cookieName: string): string {
```
Comment (danylott) [inline src/index.ts:56]: Maybe use smth like https://www.npmjs.com/package/cookie
For easier way to parse things?

File: src/index.ts [hunk 4]
```diff
@@ -47,36 +46,84 @@ export async function getApiKey(
...
+
+  setCookie(cookieName, result.visitorId, expiresDate);
+
+  return result.visitorId;
+}
+
+async function getFingerPrintResult(recureApiKey: string): Promise<any> {
+    const fp: any = await FingerprintJS.load({ apiKey: recureApiKey });
+
+    return await fp.get()
```
Comment (danylott) [inline src/index.ts:88]: ;
Use prettier

**Review by danylott — CHANGES_REQUESTED** [review]

File: src/index.ts [hunk 5]
```diff
@@ -1,8 +1,9 @@
 "use strict";
 import FingerprintJS from "@fingerprintjs/fingerprintjs-pro";
 
-export const apiKeyUrl: string = 'https://api.recure.ai/api/event_handler/get_api_key/';
-export const eventHandlerUrl: string = 'https://api.recure.ai/api/event_handler/';
+const apiKeyUrl: string = 'https://api.recure.ai/api/event_handler/get_api_key/';
+const eventHandlerUrl: string = 'https://api.recure.ai/api/event_handler/';
+const expiresDate: number = 60;
```
[the remark this answers is not in the document]
Comment (Abnormaltype) [reply src/index.ts:6]: fixed

File: src/index.ts (hunk 4 above)
Comment (Abnormaltype) [reply src/index.ts:88]: fixed

File: src/index.ts (hunk 2 above)
Comment (ArtemTopolskyi) [reply src/index.ts:53]: Yep, I believe that modify document.cookie directly - is bad practice. It's quite common action, so you can just google examples

Comment (ArtemTopolskyi) [reply src/index.ts:53]: From my experience - common practice is to set cookies on API part. But you have front package and cannot do that, not sure what's the best way to do

**Review by danylott — APPROVED** [review]

File: src/index.ts [hunk 6]
```diff
@@ -1,8 +1,10 @@
 "use strict";
 import FingerprintJS from "@fingerprintjs/fingerprintjs-pro";
+import * as Cookies from 'js-cookie';
```
Comment (danylott) [inline src/index.ts:3]: Why not just this:
```suggestion
import Cookies from 'js-cookie';
```

**Review by danylott — COMMENTED** [review]
Looks much better to me

File: src/index.ts (hunk 6 above)
Comment (Abnormaltype) [reply src/index.ts:3]: image
because of that

**Pull request merged**